maven-model-builder/src/test/java/org/apache/maven/model/profile/activation/FileProfileActivatorTest.java

```diff
@@ -67,40 +67,40 @@ void testRootDirectoryWithNull() {
 
         IllegalStateException e = assertThrows(
                 IllegalStateException.class,
-                () -> assertActivation(false, newExistsProfile("${rootDirectory}"), context));
+                () -> assertActivation(false, newExistsProfile("${project.rootDirectory}"), context));
         assertEquals(RootLocator.UNABLE_TO_FIND_ROOT_PROJECT_MESSAGE, e.getMessage());
     }
 
     @Test
     void testRootDirectory() {
-        assertActivation(false, newExistsProfile("${rootDirectory}/someFile.txt"), context);
-        assertActivation(true, newMissingProfile("${rootDirectory}/someFile.txt"), context);
-        assertActivation(true, newExistsProfile("${rootDirectory}"), context);
-        assertActivation(true, newExistsProfile("${rootDirectory}/" + "file.txt"), context);
-        assertActivation(false, newMissingProfile("${rootDirectory}"), context);
-        assertActivation(false, newMissingProfile("${rootDirectory}/" + "file.txt"), context);
+        assertActivation(false, newExistsProfile("${project.rootDirectory}/someFile.txt"), context);
+        assertActivation(true, newMissingProfile("${project.rootDirectory}/someFile.txt"), context);
+        assertActivation(true, newExistsProfile("${project.rootDirectory}"), context);
+        assertActivation(true, newExistsProfile("${project.rootDirectory}/" + "file.txt"), context);
+        assertActivation(false, newMissingProfile("${project.rootDirectory}"), context);
+        assertActivation(false, newMissingProfile("${project.rootDirectory}/" + "file.txt"), context);
     }
 
     @Test
     void testIsActiveNoFile() {
         assertActivation(false, newExistsProfile(null), context);
         assertActivation(false, newExistsProfile("someFile.txt"), context);
-        assertActivation(false, newExistsProfile("${basedir}/someFile.txt"), context);
+        assertActivation(false, newExistsProfile("${project.basedir}/someFile.txt"), context);
```

Comment:
We still have a short version `basedir` in code ... so maybe we can leave at least one test for it